File: comu/com.energyict.mdc.protocol.api/src/main/java/com/energyict/mdc/protocol/api/ConnectionFunction.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.energyict.mdc.protocol.api;

import com.energyict.mdc.upl.UPLConnectionFunction;

/**
 * Connexo specific implementation of the {@link UPLConnectionFunction} interface,
 * adding methods for localization
 *
 * @author Stijn Vanhoorelbeke
 * @since 23.06.17 - 14:56
 */
public interface ConnectionFunction extends UPLConnectionFunction {

    /**
     * The display name of this ConnectionFunction
     *
     * @return the name the User will see for this ConnectionFunction
     */
    String getConnectionFunctionDisplayName();

}

File: comu/com.energyict.comserver.comserver.core.extjs/src/main/web/js/mdc/src/model/ConnectionFunction.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Mdc.model.ConnectionFunction', {
    extend: 'Ext.data.Model',
    fields: [
        {name: 'id', type: 'int', useNull: true},
        'localizedValue',
        {name: 'alreadyUsed', type: 'boolean', useNull: true}
    ]
});